server/models/procedures.sql

```diff
@@ -1445,4 +1445,113 @@ BEGIN
   WHERE
     sm.document_uuid = document_uuid;
 END $$
+
+-- This processus inserts records relative to the stock movement integration in the posting journal
+CREATE PROCEDURE PostIntegration (
+  IN document_uuid BINARY(16),
+  IN date DATETIME,
+  IN enterprise_id SMALLINT(5) UNSIGNED,
+  IN project_id SMALLINT(5) UNSIGNED,
+  IN currency_id TINYINT(3) UNSIGNED,
+  IN user_id SMALLINT(5) UNSIGNED
+)
+BEGIN
+  DECLARE InvalidInventoryAccounts CONDITION FOR SQLSTATE '45006';
+  DECLARE current_fiscal_year_id MEDIUMINT(8) UNSIGNED;
+  DECLARE current_period_id MEDIUMINT(8) UNSIGNED;
+  DECLARE current_exchange_rate DECIMAL(19, 4) UNSIGNED;
+  DECLARE transaction_id VARCHAR(100);
+  DECLARE verify_invalid_accounts SMALLINT(5);
+  DECLARE STOCK_INTEGRATION_TRANSACTION_TYPE TINYINT(3) UNSIGNED;
+
+  -- should not be reassigned during the execution, to know why 12 please see the transaction_type table
+  SET STOCK_INTEGRATION_TRANSACTION_TYPE = 12;
+
+
+  -- getting the curent fiscal year
+  SET current_fiscal_year_id = (
+    SELECT id FROM fiscal_year AS fy
+    WHERE date BETWEEN fy.start_date AND DATE(ADDDATE(fy.start_date, INTERVAL fy.number_of_months MONTH)) AND fy.enterprise_id = enterprise_id
+  );
+
+  -- getting the period id
+  SET current_period_id = (
+    SELECT id FROM period AS p
+    WHERE DATE(date) BETWEEN DATE(p.start_date) AND DATE(p.end_date) AND p.fiscal_year_id = current_fiscal_year_id
+  );
+
+  -- getting the transaction number
+  SET transaction_id = GenerateTransactionId(project_id);
```

Comment:
This should probably be done after the check for invalid accounts.  That way, it is declared closer to where it is used (and easier to find), and we are less likely to generate duplicate transaction IDs.